### PR TITLE
Don't duplicate ImageSets when an Image with the same name attempts to be created twice

### PR DESCRIPTION
**Pull request by maxamillion**
Signed-off-by: Adam Miller <admiller@redhat.com>

# Description

Removes the ability to create a duplicate an ImageSet.

## Type of change

What is it?

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [-] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I run `go fmt ./...` to check that my code is properly formatted
- [x] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

**Review by Avielyo10 — APPROVED** [review]
ltgm

File: pkg/services/images.go [hunk 1]
```diff
@@ -75,15 +75,34 @@ type ImageService struct {
...
+		Find(&imageSetExists).
+		Error
+
+	if (err == nil) && imageSetExists {
+		result := db.DB.Where("name = ? AND account = ?", image.Name, account).First(&imageSet)
+		if result.Error != nil {
+			s.log.WithField("error", result.Error.Error()).Error("Error checking for previous image set existence")
+			return result.Error
+		}
+		log.Errorf("ImageSet %s already exists, UpdateImage transaction expected and not CreateImage", image.Name)
```
Comment (biancarosa) [inline pkg/services/images.go:95]:
```suggestion
		s.log.WithField("imageSetName", name).Error("ImageSet already exists, UpdateImage transaction expected and not CreateImage", image.Name)
```

**Review by biancarosa — CHANGES_REQUESTED** [review]

File: pkg/services/images.go [hunk 2]
```diff
@@ -75,15 +75,34 @@ type ImageService struct {
...
-	imageSet.Account = account
-	imageSet.Name = image.Name
+
+	err := db.DB.Model(imageSetModel).
+		Select("count(*) > 0").
+		Where("name = ? AND account = ?", image.Name, account).
+		Find(&imageSetExists).
+		Error
+
+	if (err == nil) && imageSetExists {
```
Comment (biancarosa) [inline pkg/services/images.go:89]: shouldn't we check if err != nil before that block? if returning an image fails i think we should give an internal error, no?

**Comment by Avielyo10** [comment]
@Mergifyio refresh

File: pkg/services/images.go (hunk 2 above)
Comment (maxamillion) [reply pkg/services/images.go:89]: oh, yes ... .yes we should, thanks!

**Comment by maxamillion** [comment]
@Mergifyio refresh

**Pull request merged**